### PR TITLE
[Clang] [Test] Use lit Syntax for Environment Variables in Clang subproject

### PR DESCRIPTION
**Pull request by Harini0924**
This patch updates the clang tests by replacing shell command substitutions with lit-compatible syntax for setting and referencing environment variables. Specifically, the use of shell-style variable substitution (e.g., `DEFAULT_TRIPLE=`and `EXPECTED_RESOURCE_DIR=`) has been replaced with `env` and `%{env}` to align with lit's internal shell requirements. These changes ensure that environment variables are properly set and accessed within the lit environment.

When using the lit internal shell with the command `LIT_USE_INTERNAL_SHELL=1 ninja check-clang`, one common error encountered is:
```
FAIL: Clang :: Driver/program-path-priority.c (19 of 20640)
******************** TEST 'Clang :: Driver/program-path-priority.c' FAILED ********************
Exit Code: 127

Command Output (stdout):
--
# RUN: at line 90
DEFAULT_TRIPLE=`/usr/local/google/home/harinidonthula/llvm-project/build/tools/clang/test/Driver/Output/program-path-priority.c.tmp/clang --version | grep "Target:" | cut -d ' ' -f2`
# executed command: 'DEFAULT_TRIPLE=`/usr/local/google/home/harinidonthula/llvm-project/build/tools/clang/test/Driver/Output/program-path-priority.c.tmp/clang' --version
# .---command stderr------------
# | 'DEFAULT_TRIPLE=`/usr/local/google/home/harinidonthula/llvm-project/build/tools/clang/test/Driver/Output/program-path-priority.c.tmp/clang': command not found
# `-----------------------------
# error: command failed with exit status: 127
```
To fix this issue, the patch replaces traditional shell substitutions with lit's environment variable handling, ensuring compatibility with the lit internal shell framework. This update applies to both the handling of the `DEFAULT_TRIPLE` and `EXPECTED_RESOURCE_DIR` variables, allowing the tests to pass when using the lit internal shell.
The patch also adds `env` to the `PWD` variable setting in the following command to ensure the environment variable is correctly set within the lit internal shell:
```
// RUN: %if system-linux %{ env PWD=/proc/self/cwd %clang -### -c --coverage %s -o foo/bar.o 2>&1 | FileCheck --check-prefix=PWD %s %}
```
[link to RFC](https://discourse.llvm.org/t/rfc-enabling-the-lit-internal-shell-by-default/80179)

### CODE REVIEW

**Comment by Harini0924** [comment]
CC: @ilovepi @petrhosek

File: clang/test/Driver/program-path-priority.c [hunk 1]
```diff
@@ -87,12 +87,13 @@
...
-// RUN: env "PATH=%t/env/" %t/clang -### -target notreal-none-elf %s 2>&1 | \
-// RUN:   FileCheck --check-prefix=DEFAULT_TRIPLE_GCC %s
-// DEFAULT_TRIPLE_GCC: env/notreal-none-elf-gcc"
+// RUN: %clang --version | grep "Target:" | cut -d ' ' -f2 > %t/default-triple.txt 
+// RUN: env DEFAULT_TRIPLE=%{readfile:%t/default-triple.txt} 
+// RUN: touch %t/%{env:DEFAULT_TRIPLE}-gcc && chmod +x %t/%{env:DEFAULT_TRIPLE}-gcc 
+// RUN: touch %t/%target_triple-gcc && chmod +x %t/%target_triple-gcc && 
+// RUN: env "PATH=%t:$PATH" %t/clang -### -target notreal-none-elf %s 2>&1 | \
+// RUN: FileCheck --check-prefix=DEFAULT_TRIPLE_GCC %s
+// DEFAULT_TRIPLE_GCC: "gcc" "-o" "a.out"
```
Comment (ilovepi) [inline clang/test/Driver/program-path-priority.c:96]: Why are these lines different? I don't think they should be changed. At the very least, the check shouldn't have been updated.

File: clang/test/Driver/program-path-priority.c [hunk 2]
```diff
@@ -87,12 +87,13 @@
...
-// RUN: touch %t/$DEFAULT_TRIPLE-gcc && chmod +x %t/$DEFAULT_TRIPLE-gcc
-// RUN: touch %t/%target_triple-gcc && chmod +x %t/%target_triple-gcc
-// RUN: env "PATH=%t/env/" %t/clang -### -target notreal-none-elf %s 2>&1 | \
-// RUN:   FileCheck --check-prefix=DEFAULT_TRIPLE_GCC %s
-// DEFAULT_TRIPLE_GCC: env/notreal-none-elf-gcc"
+// RUN: %clang --version | grep "Target:" | cut -d ' ' -f2 > %t/default-triple.txt 
+// RUN: env DEFAULT_TRIPLE=%{readfile:%t/default-triple.txt} 
+// RUN: touch %t/%{env:DEFAULT_TRIPLE}-gcc && chmod +x %t/%{env:DEFAULT_TRIPLE}-gcc 
+// RUN: touch %t/%target_triple-gcc && chmod +x %t/%target_triple-gcc && 
+// RUN: env "PATH=%t:$PATH" %t/clang -### -target notreal-none-elf %s 2>&1 | \
```
Comment (ilovepi) [inline clang/test/Driver/program-path-priority.c:94]: Why is `PATH` set differently?

Comment (Harini0924) [reply clang/test/Driver/program-path-priority.c]: Thank you for catching this. You're absolutely right—the `PATH` setting and the check shouldn't have been changed. I was experimenting with a different implementation, and I mistakenly left those changes in before committing. However, I reverted these changes back to their original state in my latest patch, which restored the correct behavior.

File: clang/test/Driver/program-path-priority.c [hunk 3]
```diff
@@ -87,8 +87,8 @@
 
 /// <default-triple>-gcc has lowest priority so <triple>-gcc
 /// on PATH beats default triple in program path
-// RUN: DEFAULT_TRIPLE=`%t/clang --version | grep "Target:" | cut -d ' ' -f2`
-// RUN: touch %t/$DEFAULT_TRIPLE-gcc && chmod +x %t/$DEFAULT_TRIPLE-gcc
+// RUN: %clang --version | grep "Target:" | cut -d ' ' -f2 > %t/default-triple.txt
```
Comment (ilovepi) [inline clang/test/Driver/program-path-priority.c:90]:
```suggestion
// RUN: %t/clang --version | grep "Target:" | cut -d ' ' -f2 > %t/default-triple.txt
```
Wasn't the test originally written that way? `%clang` is its own substitution that invokes the driver: https://llvm.org/docs/TestingGuide.html
While I don't think you'll get meaningfully different behavior, lets leave the existing behavior intact unless we have a reason to change it.

**Review by ilovepi — APPROVED** [review]
LGTM, once that last nit is fixed, assuming CI is passing.

**Pull request merged**